pkg/dockerregistry/server/admin.go

```diff
@@ -48,13 +48,9 @@ func (bh *blobHandler) Delete(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	bd, err := storage.RegistryBlobDeleter(bh.Namespace())
-	if err != nil {
-		bh.Errors = append(bh.Errors, errcode.ErrorCodeUnknown.WithDetail(err))
-		return
-	}
+	vacuum := storage.NewVacuum(bh.Context, StorageDriver)
```

Comment:
> Can't the `registry.Blobs()` be cast to `distribution.BlobDeleter` in order to ditch the global `StorageDriver`?

No. Since `registry.Blobs()` returns interface.

> By the way, when is the `StorageDriver` initialized?

The `StorageDriver` is [initialized](https://github.com/openshift/origin/pull/8938/commits/90e1d77b7e06d7b47f0f8a77a55988bcf227d796#diff-50be8483a39945647e92c49b7eb7665fR14) at server [startup.](https://github.com/docker/distribution/blob/master/registry/handlers/app.go#L148)

> Can we be sure it's initialized before a call to this handler?

Yes.